### PR TITLE
chore: upgrade all dependencies, add CODEOWNERS and SAST scanning

### PR DESCRIPTION
**Pull request by RomainMuller**
Further improve the OpenSSF score by upgrading all dependencies to reduce the amount of apparent vulnerabilities present there.

Additionally, add a `CODEOWNERS` file with obvious contents to satisfy and OpenSSF suggestion.

Finally, runs CodeQL analysis on the codebase.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+* @DataDog/apm-go @DataDog/asm-go
```
Comment (eliottness) [inline .github/CODEOWNERS:1]: Wouldn't it be better to use @dd-trace-go-guild to also include people fron profiling ?

**Review by eliottness — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (RomainMuller) [reply .github/CODEOWNERS:1]: They have a tiny, tiny integration and I would rather avoir imposing notification fatigue on them

File: go.mod [hunk 2]
```diff
@@ -1,16 +1,18 @@
 module github.com/datadog/orchestrion
 
-go 1.21
+go 1.22.0
```
Comment (eliottness) [inline go.mod:3]: I thought you did not wanted to switch to 1.22 because of Go Support Policy

Comment (RomainMuller) [reply go.mod:3]: 😡 this is `go mod tidy` doing things nobody ever asked it to 😡

**Review by eliottness — APPROVED** [review]

**Pull request merged**